### ISSUE TITLE
Can't include package by using git URL

### ISSUE THREAD

**Issue by ydGDA** [Haoming02/ollama-unity#3]
[Package Manager Window] Error adding package: https://github.com/Haoming02/ollama-unity.git.
Unable to add package [https://github.com/Haoming02/ollama-unity.git]:
  Error while cloning package from Git repository (repository: https://github.com/Haoming02/ollama-unity.git, revision: 'HEAD', package path: root):
    Repository does not contain a package manifest:
      The file [E:\UnityTrialsDec2024\OllamaUnityDemo\Library\PackageCache\.tmp-11552-QrHCNrA0j4yv\clone\package.json] cannot be found

**Comment by ydGDA** [comment]
i have cloned it by GitHub Desktop , and even tried to add it by Add Package from disk , i couldn't find package.json

**Comment by Haoming02** [comment]
Just download the `.unitypackage` from [Releases](https://github.com/Haoming02/ollama-unity/releases)

**Comment by ydGDA** [comment]
i have installed the package but when i run demo , its gives me error "WebException: The remote server returned an error: (404) Not Found." 
i have started ollama and changed the model to phi3 instead of gemma2:2b

**Comment by ydGDA** [comment]
in Request.cs in 
var httpResponse = await httpWebRequest.GetResponseAsync();

**Comment by Haoming02** [comment]
Just tried in a brand new project and it still works fine

Make sure `Ollama` is actually running:

<p align="center"><img src="https://github.com/user-attachments/assets/9bbee376-6a54-48c2-9fd8-2654f9fb4447"></p>

**Comment by Haoming02** [comment]
> changed the model to `phi3`

Did you pull the model first btw?

**Comment by ydGDA** [comment]
yes i have ollama started and i have run gemma2:2b too, i can even chat with gemma2:2b on command prompt
i have checked even ollama is running , by checking "http://localhost:11434/" or "http://127.0.0.1:11434/" opened them on browser and it displayed "Ollama is running"

**Comment by Haoming02** [comment]
Ensure your model name is typed correctly I guess

Also, give the full error log

**Comment by ydGDA** [comment]
i have created another scene and in start method 
 async void Start()
 {
     var models = await Ollama.List();
     foreach (var model in models)
     { Debug.Log($"{model.name}: [{string.Join(',', model.details.families)}]"); }

     Debug.Log("finished loading models list");
 }

and it has listed 4 models i have pulled on Ollma  gemma2:2b , tinyllama,mistral, phi3

**Comment by ydGDA** [comment]
i have written it "gemma2:2b"

**Comment by ydGDA** [comment]
WebException: The remote server returned an error: (404) Not Found.
System.Net.HttpWebRequest.GetResponseFromData (System.Net.WebResponseStream stream, System.Threading.CancellationToken cancellationToken) (at <e243450672844c6f9f403082ca9f1be7>:0)
System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) (at <e243450672844c6f9f403082ca9f1be7>:0)
System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) (at <e243450672844c6f9f403082ca9f1be7>:0)
System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) (at <ed969b0e627d471da4848289f9c322df>:0)
--- End of stack trace from previous location where exception was thrown ---
Ollama.PostRequest[T] (System.String payload, System.String endpoint) (at Assets/ollama-unity/Ollama/Requests.cs:44)
Ollama.Embeddings (System.String model, System.String input, Ollama+KeepAlive keep_alive) (at Assets/ollama-unity/Ollama/Embeddings.cs:19)
Ollama.AppendData (System.String data) (at Assets/ollama-unity/Ollama/RAG.cs:79)
OllamaRAGDemo.Start () (at Assets/ollama-unity/Demo/OllamaRAGDemo.cs:24)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) (at <ed969b0e627d471da4848289f9c322df>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at <44f3679c53d1477a9c6e72f269e3a3a9>:0)
UnityEngine.UnitySynchronizationContext.Exec () (at <44f3679c53d1477a9c6e72f269e3a3a9>:0)
UnityEngine.UnitySynchronizationContext.ExecuteTasks () (at <44f3679c53d1477a9c6e72f269e3a3a9>:0)

**Comment by Haoming02** [comment]
Welp, still can't repro this...

Have you tried the Chat and Generate Demo btw?

**Comment by ydGDA** [comment]
well , i have select Demo scene and run , once i run it gives me this error

**Comment by Haoming02** [comment]
Also, if your `Ollama` installation up-to-date?

**Comment by ydGDA** [comment]
yes i have updated it

**Comment by Haoming02** [comment]
> well , i have select Demo scene and run , once i run it gives me this error

There are 3 demos in the Scene, try the others

**Comment by ydGDA** [comment]
i have ollama version 0.5.4

**Comment by ydGDA** [comment]
I am selecting Demo scene then i press run

**Comment by ydGDA** [comment]
i have changed the model name in RAG Demo and it run without errors but alptop Fan was so loud and nothing was displayed , just loading and rectangle kept moving right and left

**Comment by ydGDA** [comment]
I have unchecked the RAG Demo , run Demo scene , it ran but nothing was displayed just loading and white rectangle kept moving right and left

**Comment by Haoming02** [comment]
> I have unchecked the RAG Demo , run Demo scene , it ran but nothing was displayed just loading and white rectangle kept moving right and left

Well, did you enable another Demo? Cause if not, then obviously there is nothing.

**Comment by Haoming02** [comment]
> I have unchecked the RAG Demo , run Demo scene , it ran but nothing was displayed just loading and white rectangle kept moving right and left

Disable and enable the entire GameObject, not just the script...

**Comment by ydGDA** [comment]
ok i have unchecked RAG game object , and only checked Generate and chat game objects . It worked but nothing is displayed only the box keep going left and right 
I have pressed 1+1 it took some time to display Answer=2

**Comment by ydGDA** [comment]
Generate Demo and chat demo worked but RAG Demo is not working , just display Loading nothing appears

**Comment by Haoming02** [comment]
Cause it's generating the embeddings at the beginning

For reference, on a RTX 3060 it took less than 10 seconds to finish loading

**Comment by ydGDA** [comment]
I have laptop with RTX 3070 Ti

**Comment by ydGDA** [comment]
the problem is that Input field is not clickable and when i press enter it don't execute the method OnSubmit in RAGDemo, although binding is set in Inspector

**Comment by ydGDA** [comment]
i have debugged the code , in Start method of OllamaRAGDemo.cs , it only execute Ollama.InitRAG(demoModel); after that data is not appended

**Comment by ydGDA** [comment]
it doesn't give exceptions , just nothing appears and it keeps looping and don't reach display.text = "ready";

**Comment by Haoming02** [comment]
Laptop GPU is more or less a generation behind equivalent desktop GPU anyway

**Comment by Haoming02** [comment]
Try only 1 TextAsset if you still want to do it
